tree/ntuple/v7/src/RNTupleModel.cxx

```diff
@@ -318,6 +318,50 @@ void ROOT::Experimental::RNTupleModel::AddField(std::unique_ptr<RFieldBase> fiel
    fFieldZero->Attach(std::move(field));
 }
 
+void ROOT::Experimental::RNTupleModel::AddSubfield(std::string_view qualifiedFieldName, REntry &entry,
+                                                   bool initializeValue) const
+{
+   auto field = FindField(qualifiedFieldName);
+   if (initializeValue)
+      entry.AddValue(field->CreateValue());
+   else
+      entry.AddValue(field->BindValue(nullptr));
+}
+
+void ROOT::Experimental::RNTupleModel::RegisterSubfield(std::string_view qualifiedFieldName)
+{
+   if (qualifiedFieldName.empty())
+      throw RException(R__FAIL("no field name provided"));
+
+   if (fRegisteredSubfields.find(std::string(qualifiedFieldName)) != fRegisteredSubfields.end())
+      throw RException(R__FAIL("subfield already registered"));
+
+   EnsureNotFrozen();
+
+   auto *field = static_cast<ROOT::Experimental::RFieldBase *>(fFieldZero.get());
+   for (auto subfieldName : ROOT::Split(qualifiedFieldName, ".")) {
+      const auto subfields = field->GetSubFields();
+      auto it = std::find_if(subfields.begin(), subfields.end(),
+                             [&](const auto *f) { return f->GetFieldName() == subfieldName; });
+      if (it != subfields.end()) {
+         field = *it;
+      } else {
+         field = nullptr;
+         break;
+      }
+      if (field->GetStructure() == ENTupleStructure::kCollection) {
+         throw RException(R__FAIL("registering a subfield in a collection is not supported"));
+      }
+   }
```

Comment:
This duplicates logic from `FindField`... An alternative would be better to just call that and then have an easier loop traversing `RFieldBase::GetParent()`, not sure